swift/ql/lib/codeql/swift/controlflow/internal/ControlFlowElements.qll

```diff
@@ -41,6 +41,13 @@ predicate isPropertyGetterElement(PropertyGetterElement pge, Accessor accessor,
   pge = TPropertyGetterElement(accessor, ref)
 }
 
+predicate isNormalAutoClosureOrExplicitClosure(ClosureExpr clos) {
+  clos instanceof AutoClosureExpr and
+  exists(CallExpr call | call.getAnArgument().getExpr() = clos)
```

Comment:
Why do we check that the `AutoClosureExpr` in an argument to a call?  When wouldn't it be?

Reply:
Short-circuiting logical operators and nil-coalescing use autoclosure exprs also.

Reply:
Makes sense, thanks.

Reply:
To be completely clear for future readers, logical operators _are_ functions and they do have _arguments_, but crucially for this code they are not called by a `CallExpr`.  They are "called" by a `BinaryExpr` and that's the case we're excluding here.